client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -376,6 +379,17 @@ public Builder setDefaultDatabase(String database) {
             return this;
         }
 
+        public Builder addProxy(ProxyType type, String host, int port) {
```

Comment:
I think the main point of adding a proxy is setting credential for it. 
Another usecase is to let client understand that real server name differs from the target where request is sent. 
Lets discuss it on the standup.